hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/metrics/FlinkCompactionMetrics.java

```diff
@@ -110,16 +110,18 @@ public void endCompaction() {
     this.compactionCost = stopTimer(COMPACTION_KEY);
   }
 
-  public void setCompactionFailedState(CompactionState compactionState){
-    this.compactionFailedState = compactionState.state;
+  public void markCompactionCompleted(CompactionState compactionState) {
+    this.compactionStateSignal = compactionState.state;
+  }
```

Comment:
There is no need to pass around `CompactionState`, just set up the constant 1 or 0 internally, the `CompactionState` can be eliminated.